File: leetcode/js/clear-digits.js
/**
 * @param {string} s
 * @return {string}
 */
var clearDigits = function (s) {
  const stack = [];

  for (const c of s) {
    const digit = parseInt(c, 10);

    if (!isNaN(digit)) {
      stack.pop();
    } else {
      stack.push(c);
    }
  }

  return stack.join("");
};

File: leetcode/rust/clear-digits.rs
impl Solution {
    pub fn clear_digits(s: String) -> String {
        let mut stack = Vec::new();

        for c in s.chars() {
            match c.is_digit(10) {
                true => {
                    stack.pop();
                }
                false => {
                    stack.push(c);
                }
            }
        }

        stack.iter().collect::<String>()
    }
}
